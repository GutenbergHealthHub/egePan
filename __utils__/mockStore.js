
// (C) Copyright IBM Deutschland GmbH 2021.  All rights reserved.

/***********************************************************************************************
imports
***********************************************************************************************/

// redux
/*-----------------------------------------------------------------------------------*/

import thunk from 'redux-thunk'
import { combineReducers, createStore, applyMiddleware } from 'redux'

// reducers
/*-----------------------------------------------------------------------------------*/

import LoginReducer from '../src/screens/login/loginReducer'
import AboutReducer from '../src/screens/about/aboutReducer'
import CheckInReducer from '../src/screens/checkIn/checkInReducer'

// services
/*-----------------------------------------------------------------------------------*/

import localStorage from '../src/services/localStorage/localStorage'

/***********************************************************************************************
reducer
***********************************************************************************************/

const appReducer = combineReducers({
	Login: LoginReducer,
	CheckIn: CheckInReducer,
	About: AboutReducer
})

const rootReducer = (state, action) => {

	// global actions that need to be available anywhere,
	// will be filtered out and executed by the rootReducer
	// (before the actual reducer is)

	if (action.type === 'USER_LOGOUT') {
		state = undefined
		localStorage.removeLastSubjectId()
	}

	if (action.type === 'DELETE_ALL_LOCAL_DATA') {
		localStorage.clearAll()
	}

	if (action.type === 'DELETE_LOCAL_QUESTIONNAIRE') {
		state.CheckIn.currentCategoryIndex = null
		state.CheckIn.questionnaireItemMap = null
		state.CheckIn.categories = null
		state.CheckIn.categoriesLoaded = false
		state.CheckIn.currentPageIndex = 1

		let subjectId = state.CheckIn.user.subjectId
		localStorage.removeLastQuestionnaireId(subjectId)
		localStorage.removeCategories(subjectId)
		localStorage.removeQuestionnaireItemMap(subjectId)
	}

	return appReducer(state, action)
}

/***********************************************************************************************
middleware
***********************************************************************************************/

const middleware = []
middleware.push(thunk)

// if(__DEV__) middleware.push(createLogger({collapsed: true}))

/***********************************************************************************************
export
***********************************************************************************************/

let globalMockStore

export function buildStore(state) {
	if(state) globalMockStore = createStore(rootReducer, state, applyMiddleware(...middleware))
	if(!state) globalMockStore = createStore(rootReducer, applyMiddleware(...middleware))
	
	return globalMockStore
}

export default buildStore()