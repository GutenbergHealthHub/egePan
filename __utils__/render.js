import React from 'react'
import { Provider } from 'react-redux'
import { render } from '@testing-library/react-native'
import { buildStore } from './mockStore'

let store = null
 
export function renderWithRedux(subComponent, state) {

    if(!store) {
        state = state || {}
        store = buildStore(state)
    }

    const queries = render(<Provider store={store}>{subComponent}</Provider>)
    
    return {
        ...queries,
        store,
    }
}

export function initRenderModule(initialState) {

    initialState = initialState || {}
    
    store = buildStore(initialState)

    return store
}