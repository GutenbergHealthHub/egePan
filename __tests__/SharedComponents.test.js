// (C) Copyright IBM Deutschland GmbH 2021.  All rights reserved.

/***********************************************************************************************
imports
***********************************************************************************************/

import React from 'react'
import { renderWithRedux } from '../__utils__/render'

import Banner from '../src/components/banner/banner'
import ProgressBar from '../src/components/modal/progressbar'
import RedirectModal from '../src/components/modal/redirectModal'
import QuestionnaireModal from '../src/components/modal/questionnaireModal'
import ScrollIndicatorWrapper from '../src/components/scrollIndicatorWrapper/scrollIndicatorWrapper'
import Spinner from '../src/components/spinner/spinner'

/***********************************************************************************************
tests
***********************************************************************************************/

// these are mostly render tests and can be extended any time ;D

describe('SHARED COMPONENTS:', () => {
    
    // simple render test
    test ('<Banner /> can be rendered', async () => {
    
         // renders the component
        let tree = renderWithRedux(<Banner />)
    
        // checks if the component matches the snapshot
        expect(tree).toMatchSnapshot()
    })

    // simple render test
    test ('<Progressbar /> can be rendered', async () => {
    
         // renders the component
        let tree = renderWithRedux(<ProgressBar />)
    
        // checks if the component matches the snapshot
        expect(tree).toMatchSnapshot()
    })

    // simple render test
    test ('<RedirectModal /> can be rendered', async () => {

        // fake props
        let props = {
            actions: {
                hideModal: () => just.fn()
            },
            modalLink: {
                title: 'test-string'
            }
        }
    
         // renders the component
        let tree = renderWithRedux(<RedirectModal {...props}/>)
    
        // checks if the component matches the snapshot
        expect(tree).toMatchSnapshot()
    })

    // simple render test
    test ('<QuestionnaireModal /> can be rendered', async () => {

         // renders the component
        let tree = renderWithRedux(<QuestionnaireModal/>)
    
        // checks if the component matches the snapshot
        expect(tree).toMatchSnapshot()
    })

    // simple render test
    test ('<ScrollIndicatorWrapper /> can be rendered', async () => {

         // renders the component
        let tree = renderWithRedux(<ScrollIndicatorWrapper/>)
    
        // checks if the component matches the snapshot
        expect(tree).toMatchSnapshot()
    })

    // simple render test
    test ('<Spinner /> can be rendered', async () => {

         // renders the component
        let tree = renderWithRedux(<Spinner/>)
    
        // checks if the component matches the snapshot
        expect(tree).toMatchSnapshot()
    })
})
