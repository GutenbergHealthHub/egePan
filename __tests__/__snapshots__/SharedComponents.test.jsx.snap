// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SHARED COMPONENTS: <Banner /> can be rendered 1`] = `
<View
  style={
    Object {
      "alignContent": "stretch",
      "alignItems": "stretch",
      "backgroundColor": "#FFFFFF",
      "flex": 1,
      "flexDirection": "column",
      "justifyContent": "flex-start",
      "paddingTop": 30,
      "width": "100%",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "flex-start",
        "flexDirection": "row",
        "flexWrap": "wrap",
        "justifyContent": "space-between",
        "paddingTop": 10,
        "width": "100%",
        "zIndex": 2,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "flexWrap": "wrap",
          "justifyContent": "flex-end",
          "marginLeft": 15,
          "marginRight": 15,
          "textAlign": "center",
        }
      }
    >
      <Text
        accessibilityRole="header"
        numberOfLines={1}
        style={
          Object {
            "alignContent": "center",
            "alignSelf": "center",
            "color": "#1F2532",
            "flex": 1,
            "fontFamily": "IBMPlexSans-SemiBold",
            "fontSize": 33.75,
            "lineHeight": 43.125,
            "textAlign": "center",
            "width": "100%",
          }
        }
      />
      <View
        style={
          Array [
            Object {
              "borderRadius": 3,
              "overflow": "hidden",
            },
            Object {
              "borderRadius": 3,
            },
            undefined,
            false,
          ]
        }
      >
        <View
          accessibilityHint="Öffnet eine neue Seite mit zusätzlichen Informationen und Links"
          accessibilityLabel="Zusätzliche Informationen"
          accessibilityRole="button"
          accessibilityState={
            Object {
              "busy": false,
              "disabled": false,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "height": 44,
              "opacity": 1,
              "width": 104.65116279069767,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "transparent",
                "borderColor": "#2089dc",
                "borderRadius": 3,
                "borderWidth": 0,
                "flexDirection": "row",
                "justifyContent": "center",
                "padding": 8,
              }
            }
          >
            <View
              style={
                Object {
                  "overflow": "hidden",
                }
              }
            >
              <View>
                <View
                  style={
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "transparent",
                      "justifyContent": "center",
                    }
                  }
                >
                  <Text
                    allowFontScaling={false}
                    selectable={false}
                    style={
                      Array [
                        Object {
                          "color": "#a5353e",
                          "fontSize": 48.83720930232558,
                        },
                        Object {
                          "backgroundColor": "transparent",
                        },
                        Object {
                          "fontFamily": "Material Design Icons",
                          "fontStyle": "normal",
                          "fontWeight": "normal",
                        },
                        Object {},
                      ]
                    }
                    testID="iconIcon"
                  >
                    󰍜
                  </Text>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
      <Text
        accessibilityRole="header"
        numberOfLines={1}
        style={
          Object {
            "alignSelf": "center",
            "color": "#1F2532",
            "fontFamily": "IBMPlexSans",
            "fontSize": 30,
            "lineHeight": 39.375,
            "marginBottom": 8.720930232558139,
            "marginTop": -26.162790697674417,
            "position": "relative",
            "textAlign": "center",
            "width": "100%",
          }
        }
      />
    </View>
  </View>
  <Image
    resizeMode="contain"
    source={
      Object {
        "testUri": "../../../src/CUSTOMIZATION/images/logo.png",
      }
    }
    style={
      Object {
        "alignSelf": "center",
        "flex": 1,
        "marginVertical": 5,
        "position": "relative",
      }
    }
  />
</View>
`;

exports[`SHARED COMPONENTS: <Progressbar /> can be rendered 1`] = `
<View
  style={
    Object {
      "alignSelf": "center",
      "backgroundColor": "#DADADA",
      "borderRadius": 2.5,
      "height": 5,
      "marginHorizontal": 5,
      "width": "95%",
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#376991",
        "borderRadius": 2.5,
        "height": 5,
        "width": "NaN%",
      }
    }
  />
</View>
`;

exports[`SHARED COMPONENTS: <QuestionnaireModal /> can be rendered 1`] = `<View />`;

exports[`SHARED COMPONENTS: <RedirectModal /> can be rendered 1`] = `
<Modal
  animationType="none"
  deviceHeight={null}
  deviceWidth={null}
  hardwareAccelerated={false}
  hideModalContentWhileAnimating={false}
  onBackdropPress={[Function]}
  onModalHide={[Function]}
  onModalWillHide={[Function]}
  onModalWillShow={[Function]}
  onRequestClose={[Function]}
  onSwipeComplete={[Function]}
  panResponderThreshold={4}
  scrollHorizontal={false}
  scrollOffset={0}
  scrollOffsetMax={50}
  scrollTo={[Function]}
  statusBarTranslucent={false}
  supportedOrientations={
    Array [
      "portrait",
      "landscape",
    ]
  }
  swipeThreshold={100}
  transparent={true}
  visible={false}
/>
`;

exports[`SHARED COMPONENTS: <ScrollIndicatorWrapper /> can be rendered 1`] = `
<View
  onLayout={[Function]}
  style={
    Object {
      "flex": 1,
      "flexDirection": "column",
      "width": "100%",
    }
  }
>
  <RCTScrollView
    contentContainerStyle={
      Object {
        "flexGrow": 1,
      }
    }
    enableOnAndroid={true}
    innerRef={[Function]}
    onScroll={[Function]}
    style={
      Object {
        "flex": 1,
        "flexDirection": "column",
        "width": "100%",
      }
    }
  >
    <View />
  </RCTScrollView>
</View>
`;

exports[`SHARED COMPONENTS: <Spinner /> can be rendered 1`] = `
<Modal
  hardwareAccelerated={false}
  visible={true}
>
  <View
    style={
      Object {
        "backgroundColor": "#376991",
        "bottom": 0,
        "flex": 1,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "bottom": 0,
          "justifyContent": "center",
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    >
      <ActivityIndicator
        color="#a5353e"
        size="large"
        style={
          Object {
            "flex": 1,
          }
        }
      />
    </View>
  </View>
</Modal>
`;
